src/ra_server.erl

```diff
@@ -469,55 +469,60 @@ handle_leader({PeerId, #append_entries_reply{success = false,
               State0 = #{cfg := #cfg{log_id = LogId} = Cfg,
                          cluster := Nodes, log := Log0}) ->
```

Comment:
Instead of changing all the code below just add a `is_map_key(PeerId, Nodes)` guard here then add a new clause below to handle the case where the peer is not known.

Reply:
Fixed in https://github.com/rabbitmq/ra/pull/432/commits/1c69466ff392f4f1448f8cc769f90e58d3e5e4a4. Thank you for your comment.